scipy/optimize/_linprog_ip.py

```diff
@@ -87,7 +89,14 @@ def _get_solver(M, sparse=False, lstsq=False, sym_pos=True,
                 def solve(r, sym_pos=False):
                     return sps.linalg.lsqr(M, r)[0]
             elif cholesky:
-                solve = cholmod(M)
+                try:
```

Comment:
Do you say that `cholmod_analyze(M)` will typically only be needed once per problem (so very infrequently)? If so, then I agree with this logic.

Reply:
I am not very familiar with the algorithm, but after skimming through the MOSEK paper I found the following mentioned: 

> The main advantage of this approach is that numerical pivoting is not required to secure numerical stability [19]. Therefore, the pivot order can be chosen to minimize fill-in in the Cholesky decomposition. Moreover, the pivot order can be chosen once and for all, because the sparsity pattern of M is independent of D.

and

>  First, a symmetric ordering of the rows and columns of M is chosen such that the fill-in in the
> Cholesky decomposition L is limited. The problem of choosing an ordering,
> which minimizes the amount of fill-in is an NP-hard problem. Therefore, the
> problem can only be solved approximately in a reasonable amount of time. Second, when an ordering has been determined the data structure for the Cholesky
> decomposition is setup. This is referred to as the symbolic phase because no
> numerical computations are involved. Finally, the actual numerical factorization takes place using the data-structures computed in the symbolic phase. The
> ordering and symbolic phases are _only executed once_ before the first numerical
> factorization

The code would raise an exception in two cases. 1) if the factor object hasn't been initialized, or 2) if the sparsity pattern of the matrix changes (e.g. due to the user solving another problem).